File: Vanilla/sources/zomboid/zombie/radio/StorySounds/DataPoint.java
package zombie.radio.StorySounds;

public final class DataPoint {
   protected float time = 0.0F;
   protected float intensity = 0.0F;

   public DataPoint(float var1, float var2) {
      this.setTime(var1);
      this.setIntensity(var2);
   }

   public float getTime() {
      return this.time;
   }

   public void setTime(float var1) {
      if (var1 < 0.0F) {
         var1 = 0.0F;
      }

      if (var1 > 1.0F) {
         var1 = 1.0F;
      }

      this.time = var1;
   }

   public float getIntensity() {
      return this.intensity;
   }

   public void setIntensity(float var1) {
      if (var1 < 0.0F) {
         var1 = 0.0F;
      }

      if (var1 > 1.0F) {
         var1 = 1.0F;
      }

      this.intensity = var1;
   }
}

File: Vanilla/sources/zdoc/media/lua/shared/Library/DataPoint.lua
---@class DataPoint : zombie.radio.StorySounds.DataPoint
---@field protected time float
---@field protected intensity float
DataPoint = {}

---@public
---@return float
function DataPoint:getTime() end

---@public
---@return float
function DataPoint:getIntensity() end

---@public
---@param arg0 float
---@return void
function DataPoint:setIntensity(arg0) end

---@public
---@param arg0 float
---@return void
function DataPoint:setTime(arg0) end
